### PR TITLE
Provide diagnostic option to enable on-demand schema loading

### PR DESCRIPTION
**Pull request by nesterone**
VSCode supports on-demand loading of url content for `$schema`, it's a part of 
`Microsoft/vscode-json-languageservice`, however, it's not available through monaco-editor

Such functionality would be a good plus, for example it was requested in vega/editor/issues/33 and I see similar feature request here Microsoft/monaco-editor/issues/365

### CODE REVIEW

**Comment by nesterone** [comment]
Hi, @aeschli 
Any comments on this? 
I'm open to providing any correction to push that feature ;-)

**Comment by aeschli** [comment]
Oh, I see, `fetch` now works in web workers on Chrome. It would be cool to also have this working in other browsers. 
Other than that the PR looks fine, I added some remarks.

File: src/monaco.d.ts [hunk 1]
```diff
@@ -30,6 +30,10 @@ declare module monaco.languages.json {
              */
             readonly schema?: any;
         }[];
+        /**
+         *  If set, the schema service would load schema content on-demand with 'fetch' if available, otherwise content would be taken from `schemas` property
+         */
```
Comment (aeschli) [inline src/monaco.d.ts:35]: Actually, the code will always check the schemas provided through the `schemas` property first, and if nothing found, will use the requestService

**Comment by nesterone** [comment]
Hi, @aeschli  
Can't find any remarks from you

File: src/monaco.d.ts (hunk 1 above)
Comment (nesterone) [reply src/monaco.d.ts:35]: Hi, @aeschli  
I've removed part about existing functionality

**Comment by nesterone** [comment]
Hi, All
Any plans to merge PR ?

**Comment by rebornix** [comment]
Gentle ping for @aeschli

**Comment by aeschli** [comment]
@nesterone Can you remove the version change from package.json? We will do that when releasing the next editor version.

**Comment by aeschli** [comment]
Thanks @kuzya-zz !

**Comment by themez** [comment]
I've met a problem that the schema resource needs credentials, can I set a custom `schemaRequestService`?
```
schemaRequestService = function (url) { 
  return fetch(url,{
    credentials: 'include'
  }).then(response => response.text())};
```

**Pull request merged**